### PR TITLE
Task/plm 299 add strict mode

### PR DESCRIPTION
**Pull request by kilikkuo**
task(PLM-299): Add strict mode support
[Why]
1. It can be helpful to provide this mode to avoid passing in unexpected key/values for certain APIs.
2. No need to build an error if pass the validation

[How]
1. Add a dsv_feature spec class decorator
2. Parse the spec feature in SpecValidator
3. Add a test case for that
4. Revamp the return value logic & add type hints

[PLM-299]

Checks
--------------
- [x] Unit tests are included or not applicable.

[PLM-299]: https://hardcoretech.atlassian.net/browse/PLM-299?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: data_spec_validator/spec/features.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+    setattr(cls, _FEAT_PARAMS, _DSVFeatureParams(strict))
+
+    return cls
+
+
+def dsv_feature(strict=False) -> type:
+    def wrap(*args):
+        cls = args[0]
+        return _process_class(cls, strict)
```
Comment (cmj0121) [inline data_spec_validator/spec/features.py]: how about  `def wrap(cls):` or `def wrap(cls, *args):`

I thinks it would be more clear

**Review by cmj0121 — APPROVED** [review]
LGTM

File: data_spec_validator/spec/features.py (hunk 1 above)
Comment (kilikkuo) [reply data_spec_validator/spec/features.py]: Good point! Thanks

Comment (kilikkuo) [reply data_spec_validator/spec/features.py]: addressed

**Pull request merged**